libfwupdplugin/fu-context.c

```diff
@@ -1644,7 +1644,12 @@ fu_context_get_default_esp(FuContext *ctx, GError **error)
 
 			/* ignore a partition that claims to be a recovery partition */
 			name = fu_volume_get_partition_name(esp);
-			if (name != NULL) {
+			if (name == NULL)
+				name = fu_volume_get_block_name(esp);
+			if (name == NULL) {
+				g_warning("failed to get block name");
+				continue;
+			} else {
```

Comment:
I don't think Coverity is going to like this; maybe leave the `if (name != NULL) {` rather than continuing?